### PR TITLE
Feature/no cors

### LINKED ISSUE #94: Create a separte entdpoint for nodes and get rid of cors
Currently we are using CORS to access the customer cluster to get the details of the nodes
e.g. 

```
https://l308esocxc.gke.k.loodse.com/api/v1/nodes
```

Create a nodes endpoint on the api server and proxy the traffic to the customer cluster api server

```
api/v1/dc/{dc}/cluster/{cluster}/nodes
api/v1/dc/gke/cluster/l308esocxc/nodes
```

### PR DESCRIPTION
**Pull request by lbb**
Resolves #94

### CODE REVIEW

File: handler/handler.go [hunk 1]
```diff
@@ -13,3 +15,20 @@ func StatusOK(res http.ResponseWriter, _ *http.Request) {
...
 	return json.NewEncoder(w).Encode(response)
 }
+
+func encodeText(w http.ResponseWriter, response interface{}) (err error) {
+	rc, ok := response.(io.ReadCloser)
+	if !ok {
+		return errors.New("response does not implement io.ReadCloser")
+	}
+
+	// Dirty but metalinter won't let us build.
```
Comment (scruplelesswizard) [inline handler/handler.go:25]: Is this still an issue after @mrIncompetent updated metalinter?

File: handler/node.go [hunk 2]
```diff
@@ -40,6 +41,81 @@ func nodesEndpoint(
...
+			return nil, NewBadRequest("unknown kubernetes datacenter %q", req.dc)
+		}
+
+		// Get cluster from dc
+		c, err := kp.Cluster(req.user, req.cluster)
+		if err != nil {
+			return nil, err
+		}
+
+		// Metalinter is buggy due to a bug see: https://github.com/golang/linter/issues/46
```
Comment (scruplelesswizard) [inline handler/node.go]: Redundant comment
> // Metalinter is buggy due to https://github.com/golang/linter/issues/46

File: handler/node.go [hunk 3]
```diff
@@ -40,6 +41,81 @@ func nodesEndpoint(
...
+		clusterKubernetesURL := fmt.Sprintf(c.Address.URL + "/api/v1/nodes")
```
Comment (scruplelesswizard) [inline handler/node.go]: `clusterKubernetesURL` seems like a misleading var name in this context

File: handler/node.go [hunk 4]
```diff
@@ -40,6 +41,81 @@ func nodesEndpoint(
...
+
+		return res.Body, nil
+	}
+}
+
+func kubernetesNodeInfoEndpoint(kps map[string]provider.KubernetesProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(nodeReq)
+
+		// Get dc info
```
Comment (scruplelesswizard) [inline handler/node.go]: This block seems ripe for dropping in to a function

File: handler/node.go [hunk 5]
```diff
@@ -40,6 +41,81 @@ func nodesEndpoint(
...
+		kp, found := kps[req.dc]
+		if !found {
+			return nil, NewBadRequest("unknown kubernetes datacenter %q", req.dc)
+		}
+
+		// Get cluster from dc
```
Comment (scruplelesswizard) [inline handler/node.go]: This block also seems ripe for functioning out

File: handler/node.go [hunk 6]
```diff
@@ -40,6 +41,81 @@ func nodesEndpoint(
...
+		c, err := kp.Cluster(req.user, req.cluster)
+		if err != nil {
+			return nil, err
+		}
+
+		clusterKubernetesURL := fmt.Sprintf(c.Address.URL + fmt.Sprintf("/api/v1/nodes/%s", req.uid))
```
Comment (scruplelesswizard) [inline handler/node.go]: `clusterKubernetesURL` is a non-descriptive var name. Please revise

**Review by scruplelesswizard — DISMISSED** [review]
LGTM with a few nits.

Feel free to self merge once they are resolved

File: handler/handler.go (hunk 1 above)
Comment (mrIncompetent) [reply handler/handler.go:25]: Linting was fine: https://app.wercker.com/Kubermatic/api/runs/build/58355666d58fb901001ab438?step=5835567d30a11f0001f08e9a

Comment (lbb) [reply handler/handler.go:25]: It was erroring due to an unchecked error value.

**Comment by mrIncompetent** [comment]
Can we rather add exported functions to the Cluster struct?
```go
func (c *api.Cluster) GetKubeConfig() capi.Config {}
func (c *api.Cluster) GetKubeClient() (*kclient.Client, error) {}
```
I already have two other places where i would need this

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**